File: p2/AccountService.java
Enum TransactionType{
  Deposit=1,
  Withdraw=2 
}

class Operation
{
  public TransactionType type;
  public double amount; 
}

class AccountService //<-- root entity
{
  Stack<Operation> stack = new Stack<Operation>(); //< value object
  Account acc = new Account(); //<-- entity
  
  public void withdraw(double amount){
    acc.withdraw(amount);
    stack.push(new Operation(TransactionType.Withdraw,amount));
  }
  
  public void deposit(double amount){
    acc.deposit(amount);
    stack.push(new Operation(TransactionType.Deposit,amount))
  }
  
  public void undo(){
    Operation op = stack.pop();
    if(op.Type == TransactionType.Withdraw)
      acc.deposit(op.amount);
    if(op.Type == TransactionType.Deposit)
      acc.withdraw(op.amount);
      
  }
  
}

File: p2/AccountService.py
class AccountService:
  def __init__(self):
    self.stack = Stack()
    self.acc = Account()
  
  def withdraw(self, amount):
    acc.withdraw(amount)
    op = Operation(1,amount)
    stack.push(op)
 
  def deposit(amount):
    acc.deposit(amount)
    op = Operation(2,amount)
    stack.push(op)

  def undo():
    op = stack.pop();
    if(op.Type == 1):
      acc.deposit(op.amount)
    if(op.Type == 2):
      acc.withdraw(op.amount)
